File: ObjectAvoidance/java/src/main/java/elite/math/trig/Sine.java
package elite.math.trig;

import elite.math.trig.triangle.Hypotenuse;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Sine {
    
    /**
     *
     */
    public double sine;
    
    /**
     *
     * @param o
     * @param h
     */
    public Sine(Opposite o, Hypotenuse h){
        this.sine = (o.opposite / h.hypotenuse);
    }
    
    /**
     *
     * @param arcsin
     */
    public Sine(Arcsine arcsin){
        this.sine = (1 / arcsin.arcsine);
    }
    
    /**
     *
     * @param csc
     */
    public Sine(Cosecant csc){
        this.sine = (1 / csc.cosecant);
    }
    
    /**
     *
     * @return
     */
    public Cosecant toCosecant(){
        return new Cosecant(this);
    }
    
    /**
     *
     * @return
     */
    public Arcsine toArcsine(){
        return new Arcsine(this);
    }
}

File: ObjectAvoidance/kotlin/src/elite/math/trig/Sine.kt
package elite.math.trig

import elite.math.trig.triangle.Hypotenuse
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Sine {

    /**
     *
     */
    var sine: Double = 0.toDouble()

    /**
     *
     * @param o
     * @param h
     */
    constructor(o: Opposite, h: Hypotenuse) {
        this.sine = o.opposite / h.hypotenuse
    }

    /**
     *
     * @param arcsin
     */
    constructor(arcsin: Arcsine) {
        this.sine = 1 / arcsin.arcsine
    }

    /**
     *
     * @param csc
     */
    constructor(csc: Cosecant) {
        this.sine = 1 / csc.cosecant
    }

    /**
     *
     * @return
     */
    fun toCosecant(): Cosecant {
        return Cosecant(this)
    }

    /**
     *
     * @return
     */
    fun toArcsine(): Arcsine {
        return Arcsine(this)
    }
}